projects/extension/src/background/ConnectionManager.ts

```diff
@@ -153,6 +156,23 @@ export class ConnectionManager extends (EventEmitter as {
     return client.terminate()
   }
 
+  flushDatabases(): void {
+    this.#chainConnections.forEach(async (ch) => {
+      if (Array.from(wellKnownChains.keys()).includes(ch.chainName)) {
```

Comment:
As pointed out in the previous comment, this is not correct because you will be taking any chain-connection that's a well-known one, and there could be many...

Reply:
As replied in previous comment `we should pass the database content in each instance of the known chains`